tsl/src/bgw_policy/job.c

```diff
@@ -219,20 +221,125 @@ bgw_policy_job_requires_enterprise_license(BgwJob *job)
 {
 	license_print_expiration_warning_if_needed();
 
+	/* We deliberately have no default case, so that the compiler will warn if
+	 * we miss a job type
+	 */
 	switch (job->bgw_type)
 	{
 		case JOB_TYPE_REORDER:
-			return true;
 		case JOB_TYPE_DROP_CHUNKS:
+		case JOB_TYPE_SCHEDULED_INDEX:
 			return true;
 		case JOB_TYPE_CONTINUOUS_AGGREGATE:
+		case JOB_TYPE_VERSION_CHECK:
 			return false;
-		default:
-			elog(ERROR,
-				 "scheduler could not determine the license type for job type: \"%s\"",
-				 NameStr(job->fd.job_type));
+		case JOB_TYPE_UNKNOWN:
+		case _MAX_JOB_TYPE:
+			elog(ERROR, "invalid job type: \"%s\"", NameStr(job->fd.job_type));
 	}
-	pg_unreachable();
+	elog(ERROR,
+		 "scheduler could not determine the license type for job type: \"%s\"",
+		 NameStr(job->fd.job_type));
+}
+
+bool
+execute_scheduled_index_policy(BgwJob *job, bool fast_continue)
+{
+	int chunk_id;
+	bool started = false;
+	BgwPolicyScheduledIndex *args;
+	Hypertable *ht;
+	Chunk *chunk;
+	int32 job_id = job->fd.id;
+
+	if (!IsTransactionOrTransactionBlock())
+	{
+		started = true;
+		StartTransactionCommand();
+	}
+
+	/* Get the arguments from the scheduled_index_policy table */
+	args = ts_bgw_policy_scheduled_index_find_by_job(job_id);
+
+	if (args == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_TS_INTERNAL_ERROR),
+				 errmsg("could not run scheduled_index policy #%d because no args in policy table",
+						job_id)));
+
+	ht = ts_hypertable_get_by_id(args->fd.hypertable_id);
+
+	/* Find a chunk needing the index */
+	/* currently just use the same logic as reorder */
+	// chunk_id = get_chunk_id_needing_schedule_index(args->fd.job_id, ht);
+	chunk_id = get_chunk_id_to_reorder(args->fd.job_id, ht);
```

Comment:
need to rename function plus should probably make the number of recent chunks to skip (ie`REORDER_SKIP_RECENT_DIM_SLICES_N`) a param to the function.